src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1865,11 +1865,13 @@ TEST(SpatialPoolerTest, testSaveLoad) {
   setup(sp1, numInputs, numColumns);
 
   ofstream outfile;
-  outfile.open(filename);
+  outfile.open(filename, ifstream::binary);
+	outfile.precision(std::numeric_limits<double>::digits10 + 1);
+	outfile.precision(std::numeric_limits<float>::digits10 + 1);
```

Comment:
Should this precision be set by the serialize method?  Is this something that every user is going to need?

Reply:
> Should this precision be set by the serialize method?

This is what the saveToFile( ) serialization sets.  I guess in this test, I did not need to set it...it would default to whatever the max resolution is.  I will remove those so it will test that it does not need to be set.  The stream does need to be initialized as ifstream::binary however because it is taking the default serialization format which is BINARY.

Reply:
FYI This issue is also on lines:

https://github.com/htm-community/nupic.cpp/blob/418f51cf49ba928ec8f2487773f6f1ef8c13735d/src/test/unit/algorithms/SpatialPoolerTest.cpp#L1904-L1905

https://github.com/htm-community/nupic.cpp/blob/418f51cf49ba928ec8f2487773f6f1ef8c13735d/src/test/unit/algorithms/SpatialPoolerTest.cpp#L1935-L1936

https://github.com/htm-community/nupic.cpp/blob/418f51cf49ba928ec8f2487773f6f1ef8c13735d/src/test/unit/algorithms/SpatialPoolerTest.cpp#L1996-L1997

Reply:
> os.precision(std::numeric_limits<double>::digits10 + 1); 
os.precision(std::numeric_limits<float>::digits10 + 1); 

I removed the ones at line 1868 to confirm that they are not required.  I left in the other three places to the test can confirm that they can be specified by the user.

Reply:
Makes sense.